### ISSUE TITLE
TensorRT conversion fails with 5D input data

### ISSUE THREAD

**Issue by AlexM4** [tensorflow/tensorflow#55452]
Hi, I'm having an issue with tensorrt conversion of model which uses 3D convolutions and processes 5D input.

Code to reproduce error (I cut the model to minimal example):
```
import tensorflow as tf
import numpy as np
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Input,Conv3D,BatchNormalization,Activation

print("TensorFlow version: ",tf.version.VERSION)

src_dir='model_bn'
dst_dir='model_bn_conv'
batch=4
input_shape=(24,160,160,3)

def input_fn():
    yield [np.zeros(((batch,)+input_shape),dtype='float32')]

model=Sequential()
model.add(Input(shape=input_shape))
model.add(Conv3D(24,3))
model.add(BatchNormalization())
model.add(Activation('relu'))

model.save(src_dir)

params=tf.experimental.tensorrt.ConversionParams(precision_mode='FP32')
converter=tf.experimental.tensorrt.Converter(input_saved_model_dir=src_dir,conversion_params=params)

converter.convert()
print('Converted successfully!')

print('Running build()...')
converter.build(input_fn=input_fn)
print('Build completed')
converter.save(dst_dir)
```

The output:
```
TensorFlow version:  2.8.0
2022-03-31 11:44:04.711186: I tensorflow/core/platform/cpu_feature_guard.cc:152] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE3 SSE4.1 SSE4.2 AVX
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-03-31 11:44:05.311164: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 13608 MB memory:  -> device: 0, name: Tesla T4, pci bus id: 0000:3b:00.0, compute capability: 7.5
WARNING:tensorflow:Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.
2022-03-31 11:44:05.990618: W tensorflow/python/util/util.cc:368] Sets are not currently considered sequences, but this may change in the future, so consider avoiding using them.
2022-03-31 11:44:06.422604: I tensorflow/core/grappler/devices.cc:66] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 1
2022-03-31 11:44:06.422784: I tensorflow/core/grappler/clusters/single_machine.cc:358] Starting new session
2022-03-31 11:44:06.458670: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 13608 MB memory:  -> device: 0, name: Tesla T4, pci bus id: 0000:3b:00.0, compute capability: 7.5
2022-03-31 11:44:06.485282: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:1191] Optimization results for grappler item: graph_to_optimize
  function_optimizer: Graph size after: 30 nodes (20), 41 edges (31), time = 1.05ms.
  function_optimizer: function_optimizer did nothing. time = 0.01ms.

2022-03-31 11:44:06.529996: I tensorflow/core/grappler/devices.cc:66] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 1
2022-03-31 11:44:06.530104: I tensorflow/core/grappler/clusters/single_machine.cc:358] Starting new session
2022-03-31 11:44:06.539721: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 13608 MB memory:  -> device: 0, name: Tesla T4, pci bus id: 0000:3b:00.0, compute capability: 7.5
2022-03-31 11:44:06.551640: W tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:192] Calibration with FP32 or FP16 is not implemented. Falling back to use_calibration = False.Note that the default value of use_calibration is True.
2022-03-31 11:44:06.551665: I tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:211] [TF-TRT] not using explicit QDQ mode
2022-03-31 11:44:06.552184: W tensorflow/compiler/tf2tensorrt/segment/segment.cc:954]

################################################################################
TensorRT unsupported/non-converted OP Report:
        - NoOp -> 2x
        - Identity -> 1x
        - Placeholder -> 1x
--------------------------------------------------------------------------------
        - Total nonconverted OPs: 4
        - Total nonconverted OP Types: 3
For more information see https://docs.nvidia.com/deeplearning/frameworks/tf-trt-user-guide/index.html#supported-ops.
################################################################################

2022-03-31 11:44:06.552276: W tensorflow/compiler/tf2tensorrt/segment/segment.cc:1281] The environment variable TF_TRT_MAX_ALLOWED_ENGINES=20 has no effect since there are only 1 TRT Engines with  at least minimum_segment_size=3 nodes.
2022-03-31 11:44:06.552292: I tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc:795] Number of TensorRT candidate segments: 1
2022-03-31 11:44:06.552678: I tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc:909] Replaced segment 0 consisting of 11 nodes by TRTEngineOp_0_0.
2022-03-31 11:44:06.556775: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:1191] Optimization results for grappler item: tf_graph
  model_pruner: Graph size after: 22 nodes (-8), 33 edges (-8), time = 0.273ms.
  debug_stripper: debug_stripper did nothing. time = 0.008ms.
  layout: Graph size after: 26 nodes (4), 37 edges (4), time = 1.157ms.
  dependency_optimizer: Graph size after: 16 nodes (-10), 15 edges (-22), time = 0.228ms.
  constant_folding: Graph size after: 16 nodes (0), 15 edges (0), time = 0.564ms.
  common_subgraph_elimination: Graph size after: 13 nodes (-3), 15 edges (0), time = 0.124ms.
  TensorRTOptimizer: Graph size after: 3 nodes (-10), 2 edges (-13), time = 1.148ms.
  constant_folding: Graph size after: 3 nodes (0), 2 edges (0), time = 0.294ms.
Optimization results for grappler item: TRTEngineOp_0_0_native_segment
  model_pruner: Graph size after: 13 nodes (0), 15 edges (0), time = 0.083ms.
  debug_stripper: debug_stripper did nothing. time = 0.006ms.
  layout: Graph size after: 13 nodes (0), 15 edges (0), time = 0.4ms.
  dependency_optimizer: Graph size after: 12 nodes (-1), 13 edges (-2), time = 0.116ms.
  constant_folding: Graph size after: 12 nodes (0), 13 edges (0), time = 0.366ms.
  common_subgraph_elimination: Graph size after: 12 nodes (0), 13 edges (0), time = 0.093ms.
  TensorRTOptimizer: Graph size after: 12 nodes (0), 13 edges (0), time = 0.013ms.
  constant_folding: Graph size after: 12 nodes (0), 13 edges (0), time = 0.346ms.

Converted successfully!
Running build()...
2022-03-31 11:44:06.640977: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:446] TRTEngineOp not using explicit QDQ
2022-03-31 11:44:06.643431: I tensorflow/compiler/tf2tensorrt/common/utils.cc:100] Linked TensorRT version: 8.2.3
2022-03-31 11:44:06.643602: I tensorflow/compiler/tf2tensorrt/common/utils.cc:102] Loaded TensorRT version: 8.2.3
2022-03-31 11:44:07.618558: F ./tensorflow/compiler/tf2tensorrt/utils/trt_tensor_proxy.h:158] 'trt_tensor_' Must be non NULL
Aborted
```

Other observations:
- Removing activation layer or swapping it with BatchNorm makes script to execute without errors, but doesn't produce valid trt engine. It writes the following warning: "TF-TRT Warning: Engine creation for TRTEngineOp_0_0 failed. The native segment will be used instead. Reason: INVALID_ARGUMENT: Rank of perm for transpose does not match with that of the input."
- Removing Conv3D layer does not change the result.
- Trying to tweak conversion parameters did not help.
- Using 4D input data and 2D convolutions seem to work fine.

**System information**
(Using NVidia NGC container for Tensorflow, version 22.03)
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): **Yes**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): **Linux Ubuntu 20.04**
- TensorFlow installed from (source or binary): **binary**
- TensorFlow version (use command below): **2.8.0**
- Python version: **3.8.10**
- CUDA/cuDNN version: **11.6/8.3.3**
- GPU model and memory: **NVidia Tesla T4 16Gb**

**Comment by sushreebarsa** [comment]
@sachinprasadhs I was able to reproduce this issue on colab using TF v2.8.0 ,tf-nightly ,please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/8a0c18310a75b69758529ee9a8a327ab/55452.ipynb#scrollTo=fZq3m4XP3WxG).Thanks!

**Comment by tilakrayal** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
